Tools/machines/frontier-olcf/submit.sh

```diff
@@ -6,12 +6,155 @@
+    warpx_dim="3"
+
+else
+    # Replace this with the name of the WarpX executable
+    exe="warpx"
+
+    # Replace this with the name of the WarpX inputfile
+    inputfile="inputs"
+fi
+
+# File to redirect stdandard output to
+output="output.txt"
+
+# It may be a good idea to print the current date
+date
+
+########################################################################
+########## INSTRUCTIONS BELOW SHOULD NOT NORMALLY BE MODIFIED ##########
+########################################################################
+
+# ___ Broadcast WarpX and associated libraries to compute nodes_________
+
+if [ ! $is_pywarpx ]; then
+
+    # From the documentation:
+    # SBCAST executable from Orion to NVMe -- NOTE: ``-C nvme``
+    # is needed in SBATCH headers to use the NVMe drive
+    # NOTE: dlopen'd files will NOT be picked up by sbcast
+    sbcast --send-libs --exclude=NONE -pf ${exe} /mnt/bb/$USER/warpx
+    if [ ! "$?" == "0" ]; then
+        # CHECK EXIT CODE. When SBCAST fails, it may leave partial
+        # files on the compute nodes, and if you continue to launch srun,
+        # your application may pick up partially complete shared library files,
+        # which would give you confusing errors.
+        echo "SBCAST failed!"
+        exit 1
+    fi
+    # SBCAST sends all libraries detected by `ld` (minus any excluded),
+    # and stores them in the same directory in each node's node-local storage
+    # Any libraries opened by `dlopen` are NOT sent, since they are not
+    # known by the linker at run-time.
+    # All required libraries now reside in /mnt/bb/$USER/warpx_libs
+    export LD_LIBRARY_PATH="/mnt/bb/$USER/warpx_libs"
+    # libfabric dlopen's several libraries:
+    export LD_LIBRARY_PATH="${LD_LIBRARY_PATH}:$(pkg-config --variable=libdir libfabric)"
+    # cray-mpich dlopen's libhsa-runtime64.so and
+    # libamdhip64.so (non-versioned), so symlink on each node:
+    srun -N ${SLURM_NNODES} -n ${SLURM_NNODES} --ntasks-per-node=1 --label -D /mnt/bb/$USER/warpx_libs \
+        bash -c "if [ -f libhsa-runtime64.so.1 ]; then ln -s libhsa-runtime64.so.1 libhsa-runtime64.so; fi;
+        if [ -f libamdhip64.so.5 ]; then ln -s libamdhip64.so.5 libamdhip64.so; fi"
+    # You may notice that some libraries are still linked
+    # from /sw/frontier, even after SBCASTing.
+    # This is because the Spack-build modules use RPATH to find their dependencies.
+    # This behavior cannot be changed.
+    echo "*****ldd /mnt/bb/$USER/warpx*****"
+    ldd /mnt/bb/$USER/warpx
+    echo "*************************************"
+
+else
+    # Find out the path of the pywarpx module
+    PYWARPX_PATH=$(python3 -c "import pywarpx; print(pywarpx.__path__[0])")
+
+    if [ ${warpx_dim} == "1" ]; then
+        export WARPX_SO="warpx_pybind_1d.cpython-39-x86_64-linux-gnu.so"
+    elif [ ${warpx_dim} == "2" ]; then
+        export WARPX_SO="warpx_pybind_2d.cpython-39-x86_64-linux-gnu.so"
```

Comment:
Instead of asking the user, we could simply loop over the known names and copy if the file exists.

Reply:
We also need to broadcast the libraries used by the `.so` file:
```
 if [ $(basename $FILE) == $WARPX_SO ]; then
      sbcast --send-libs --exclude=NONE -pf $FILE /mnt/bb/$USER/pywarpx/$(basename $FILE)
```
Should we do that for all the found `.so` files?